### PR TITLE
Allow `onWillAcceptWithDetails` and `onAcceptWithDetails` callbacks in `DragTarget` to accept null in `data` field.

### LINKED ISSUE #136412: Allow `onWillAcceptWithDetails` and `onAcceptWithDetails` to accept `null` in `data` field.
As a continuation to [this](https://github.com/flutter/flutter/pull/133691#pullrequestreview-1668720955) comment, both of the newer API should allow `data` to be `null`.

### PR DESCRIPTION
**Pull request by chinmoy12c**
*List which issues are fixed by this PR. You must list at least one issue.*

Fixes: #136412

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/widgets/draggable_test.dart
Comment (chinmoy12c) [inline packages/flutter/test/widgets/draggable_test.dart:268]: Hey @Piinks, this is where the problem would arise in old codebases.

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (Piinks) [inline packages/flutter/lib/src/widgets/drag_target.dart:752]: If data is null why not just pass in null drag target details?

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (Piinks) [inline packages/flutter/lib/src/widgets/drag_target.dart:590]: Again, this has the potential to be super super breaking. *Any* customer that uses drag target details now has to update their code to handle if this is null, as none of them will have that now.

File: packages/flutter/test/widgets/draggable_test.dart
Comment (Piinks) [reply packages/flutter/test/widgets/draggable_test.dart:268]: I think there would be more than one problem, running tests though to find out

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (Piinks) [reply packages/flutter/lib/src/widgets/drag_target.dart:752]: This is what I meant by https://github.com/flutter/flutter/pull/133691#pullrequestreview-1668720955

Since the onAcceptWithDetails method is very new, it is easier to change the parameter of the function signature to be nullable. Changing the data in DragTargetDetails to be nullable is a much bigger breaking change since it has been around for a long time.

**Review by Piinks — APPROVED** [review]
Tentatively approving so I can fire off more tests internally. Please wait for confirmation to land. LGTM

**Comment by Piinks** [comment]
BTW the docs test shard is failing here

**Comment by chinmoy12c** [comment]
> BTW the docs test shard is failing here

Woops! Sorry about that, made the fix.

**Comment by Piinks** [comment]
It could be that https://github.com/flutter/flutter/pull/133136 is the best way to handle this, by just not calling the callbacks if data is null.

**Comment by Piinks** [comment]
After conferring with folks on this, I think we confirmed non-nullable data is intentional. The other PR (https://github.com/flutter/flutter/pull/133136) eliminates the crash by just not calling the callbacks when data is null. I think we should close this PR then, since we cannot change data to be null, or the function signature to take null details.

**Comment by goderbauer** [comment]
(Triage): As per the previous comment I am going to close this. Thanks you!

**Pull request closed without merging**
